compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -807,6 +807,9 @@ object SymDenotations {
                            // we need an inline flag on them only do that
                            // reduceProjection gets access to their rhs
 
+    /** Is this a Scala 2 macro */
+    final def isScala2Macro(implicit ctx: Context): Boolean = is(Macro) && symbol.owner.is(Scala2x)
```

Comment:
Why not change the Scala2Unpickler to set the Scala2x flag on methods too ?

Reply:
We could try that

Reply:
> I added an alternative fix in 4b6d417 that sets the Scala2x flag to all symbols.


That looks good to me but maybe there was a reason to not set Scala2x on all Scala 2 symbols ? /cc @odersky

Reply:
The issue is that we have to use an additional as `Scala2x` is only a type flag.

Reply:
We did not add the flag as it was not absolutely necessary